Upstream: yes

From d9783e4a1476b6787a51c5ae9e9b3156527589f0 Mon Sep 17 00:00:00 2001
From: erouault <erouault>
Date: Mon, 11 Jul 2016 21:26:03 +0000
Subject: [PATCH] * tools/tiffcrop.c: Avoid access outside of stack allocated
 array on a tiled separate TIFF with more than 8 samples per pixel. Reported
 by Kaixiang Zhang of the Cloud Security Team, Qihoo 360 (CVE-2016-5321,
 bugzilla #2558)

---
 tools/tiffcrop.c | 2 +-
 1 file changed, 1 insertions(+), 1 deletion(-)

diff --git a/tools/tiffcrop.c b/tools/tiffcrop.c
index d959ae3..6fc8fc1 100644
--- a/tools/tiffcrop.c
+++ b/tools/tiffcrop.c
@@ -989,7 +989,7 @@ static int  readSeparateTilesIntoBuffer (TIFF* in, uint8 *obuf,
     nrow = (row + tl > imagelength) ? imagelength - row : tl;
     for (col = 0; col < imagewidth; col += tw)
       {
-      for (s = 0; s < spp; s++)
+      for (s = 0; s < spp && s < MAX_SAMPLES; s++)
         {  /* Read each plane of a tile set into srcbuffs[s] */
 	tbytes = TIFFReadTile(in, srcbuffs[s], col, row, 0, s);
         if (tbytes < 0  && !ignore)
-- 
2.9.3

